### PR TITLE
Add boards to global data retention policy

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Adds Boards to Global Retention Policy UI.

#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

#### Related Pull Requests
- Has server changes 
https://github.com/mattermost/mattermost-server/pull/19262
https://github.com/mattermost/enterprise/pull/1126

#### Screenshots

Screen Shot 2021-12-29 at 2 26 00 PM

Screen Shot 2021-12-29 at 2 26 18 PM

#### Release Note

```release-note
Added Boards to Global Retention Policy

```

### CODE REVIEW

File: components/admin_console/data_retention_settings/data_retention_settings.tsx [hunk 1]
```diff
@@ -99,9 +103,23 @@ export default class DataRetentionSettings extends React.PureComponent<Props, St
...
+                    ),
+                    field: 'boards',
+                },
+            );
+        }
+        return columns;
     }
     getCustomPolicyColumns = (): Column[] => {
-        return [
+        const columns = [
```
Comment (jespino) [inline components/admin_console/data_retention_settings/data_retention_settings.tsx:122]: nit: I guess you changed it for consistency reasons, but still, the style is not consistent. I would suggest to change the other method constant initialization to the format here, I mean `const columns = [` with the `[` in the same line.

File: components/admin_console/data_retention_settings/data_retention_settings.tsx [hunk 2]
```diff
@@ -99,9 +103,23 @@ export default class DataRetentionSettings extends React.PureComponent<Props, St
                 className: 'actionIcon',
             },
         ];
+        if (this.includeBoards) {
+            columns.splice(columns.length - 1, 0,
```
Comment (jespino) [inline components/admin_console/data_retention_settings/data_retention_settings.tsx:107]: nit: This splice is a bit hard to understand in the intentions. I would just replace that with a `columns.push` here, and move the last entry of for the `actions` field, in another `column.push` at the end (right after the if). That way is easier to understand the "flow" of the data.

**Review by jespino — APPROVED** [review]
Looks good to me. A couple of tiny improvements of readability, but nothing mayor.

**Review by harshilsharma63 — APPROVED** [review]

**Review by ogi-m — APPROVED** [review]

**Pull request merged**